### PR TITLE
Refactoring of Routes, separating search and resource operations.

### PR DESCRIPTION
**Pull request by umbreak**
- Added a PathDirectives which allows to extract any resourceIds
- using the expected API structure {org}/{dom}/{name}/{ver}/{uuid}.
- Added SchemaName case class needed to represent that endpoint.

### CODE REVIEW

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/schemas/SchemaId.scala [hunk 1]
```diff
@@ -17,10 +17,12 @@ import scala.util.{Failure, Try}
   * @param name     the name of the schema
   * @param version  the version of the schema
   */
-final case class SchemaId(domainId: DomainId, name: String, version: Version)
+final case class SchemaId(domainId: DomainId, name: String, version: Version) {
+  def schemaName: SchemaName = SchemaName(domainId, name)
```
Comment (bogdanromanx) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/schemas/SchemaId.scala]: add a scaladoc here

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/schemas/SchemaName.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import scala.util.matching.Regex
+
+/**
+  * Unique schema name identifier rooted in a domain.
+  *
+  * @param domainId the domain identifier for this schema
+  * @param name     the name of the schema
+=  */
+final case class SchemaName(domainId: DomainId, name: String) {
+  def versioned(version: Version): SchemaId = SchemaId(domainId, name, version)
```
Comment (bogdanromanx) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/schemas/SchemaName.scala]: Add a scaladoc here

**Comment by umbreak** [comment]
retest this please

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala [hunk 3]
```diff
@@ -44,155 +44,126 @@ import scala.concurrent.{ExecutionContext, Future}
...
+              }
+            }
           } ~
-          pathPrefix(Segment) { domain =>
-            val domainId = DomainId(orgId, domain)
-            (pathEndOrSingleSlash & get & searchQueryParams) { (pagination, filterOpt, termOpt, deprecatedOpt) =>
-                traceName("searchInstancesOfDomain") {
-                  val filter = filterFrom(deprecatedOpt, filterOpt, querySettings.nexusVocBase)
-                  instanceQueries.list(domainId, filter, pagination, termOpt).buildResponse(base, pagination)
+          (get & traceName("getInstance")) {
```
Comment (bogdanromanx) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala]: I think it's useful to discriminate between getInstance and getInstanceRevision like before this change. Thoughts?

**Comment by umbreak** [comment]
retest this please

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala (hunk 3 above)
Comment (umbreak) [reply modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala]: It is fine. I'll then revert that block

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/schemas/SchemaName.scala (hunk 2 above)
Comment (umbreak) [reply modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/schemas/SchemaName.scala]: Done

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/schemas/SchemaId.scala (hunk 1 above)
Comment (umbreak) [reply modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/schemas/SchemaId.scala]: Done

**Comment by umbreak** [comment]
retest this please

**Pull request merged**